cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js

```diff
@@ -432,13 +433,27 @@ class HydratorPlusPlusNodeConfigCtrl {
       scope: this.$scope
     };
 
-    let connections = this.ConfigStore.getConfigForExport().config.connections;
+
+    let { stages, connections } = this.ConfigStore.getConfigForExport().config;
     let adjacencyMap = this.HydratorPlusPlusOrderingFactory.getAdjacencyMap(connections);
     let postBody = [];
-    angular.forEach(adjacencyMap, (value, key) => {
-      let inputPlugin = value.filter(node =>this.state.node.plugin.label === node);
-      if (inputPlugin.length) {
-        postBody.push(key);
+
+    angular.forEach(adjacencyMap, (outputStagesNames, inputStageName) => {
+      let clickedStageName = outputStagesNames.filter(node => this.state.node.plugin.label === node);
+      if (clickedStageName.length) {
+        let inputStage = stages.filter(stage => stage.name === inputStageName);
+        if (inputStage.length && inputStage[0].plugin.type !== 'condition') {
+          postBody.push(inputStageName);
+        } else {
+          inputStageIsCondition = true;
+          let conditionStageName = inputStageName;
+          angular.forEach(adjacencyMap, (previousOutputStageNames, previousInputStageName) => {
```

Comment:
would this be a problem if two or more condition nodes are connected to one other? Like
```
Source -> Condition 1 -- > Condition 2 -- > Sink 1
                                      |
                                       -> Condition 3 --> Transform 1 --> Sink 2
                                                                |
                                                                -> Transform 2 --> Sink 3
```

What happens if I click `Transform 1` will I be able to see the incoming data as out going data of Source?

Reply:
Yeah right now that won't work. Will fix the traversal to be recursive